WIP: Add bell icon in the CI build client to allow getting notifications

Moo64c opened the pull request:
#214

WIP.

Commit:
WIP

Commit:
Incremental.

amitaibu commented on client/app/scripts/controllers/repo.js:
```diff
@@ -12,6 +12,7 @@ angular.module('clientApp')
     $scope.build = build[0];
     $scope.ciBuildItems = [];
     $scope.incidents = incidents;
+    $scope.notificationsClass = $scope.build.notification ? 'fa-bell' : 'fa-bell-slash';
```
this should be true/ false - and in the HTML add the right class

amitaibu commented on client/app/scripts/controllers/repo.js:
```diff
@@ -67,6 +68,33 @@ angular.module('clientApp')
+    $scope.changeNotifications = function() {
+      console.log($scope);
+      var params = {
+        'toggleNotifications' : 1
```
try to use `true`

amitaibu commented on client/app/scripts/controllers/repo.js:
```diff
@@ -67,6 +68,33 @@ angular.module('clientApp')
+
+      var url = Config.backend + '/api/notification/' + $scope.build.repository;
+
+      $http
```
no $http in the controller - move to own service

amitaibu commented on shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php:
```diff
@@ -40,6 +40,22 @@ public function publicFieldsInfo() {
     );
 
+    $public_fields['notification'] = array(
+      'callback' => array($this, 'notificationProcess'),
```
move this to a new resource - OGMemershipResource. this should be a property

amitaibu commented on shoov/modules/custom/shoov_restful/plugins/restful/notification/notification/RestfulNotification.class.php:
```diff
@@ -0,0 +1,53 @@
+<?php
+
+class RestfulNotification extends \RestfulBase implements RestfulDataProviderInterface {
```
this should be `OgMembershipResource extends ResftulEntityBase`

Commit:
Add notification response.

Commit:
New version...

amitaibu commented on client/app/scripts/controllers/repo.js:
```diff
@@ -67,6 +67,31 @@ angular.module('clientApp')
+          $scope.build.notification = !$scope.build.notification;
+
+          // Hide the success icon after 3 seconds of receiving the response.
+          $timeout(function() {
```
remove

Commit:
Remove checkmark when changing notification status.

Moo64c commented:
![selection_143](https://cloud.githubusercontent.com/assets/2690042/9787662/479020ce-57cd-11e5-8ab0-d395edef2af1.png)
![selection_144](https://cloud.githubusercontent.com/assets/2690042/9787659/45ca5d22-57cd-11e5-8431-bcebcd35cf7f.png)

Moo64c committed:
Merge branch 'master' into 214

Moo64c committed:
Working flag/unflag click.

Moo64c committed:
Test start.

Moo64c committed:
getMachineName from 210

Conflicts:

shoov/modules/custom/shoov_ci_build/shoov_ci_build.features.field_base.inc
	shoov/modules/custom/shoov_ci_build/shoov_ci_build.info
	shoov/modules/custom/shoov_ci_build/shoov_ci_build.module

Moo64c committed:
Possibly better testing (that works.)(

Moo64c committed:
No behat for me I guess.

Moo64c committed:
Uhhh fix tests?

Moo64c commented:
This PR requires using https://github.com/RESTful-Drupal/restful/pull/642
Otherwise `notification` variable won't be sent to the client. It means that the notification can only be activated and the bell icon status would be wrong: (The toggle request depends on that variable).
![selection_082](https://cloud.githubusercontent.com/assets/2690042/9907469/fd68a41a-5c97-11e5-9e4a-a4a3dc3957cc.jpg)
![selection_084](https://cloud.githubusercontent.com/assets/2690042/9907524/2e79d29a-5c98-11e5-8dcc-b6ced6fbc705.jpg)

Moo64c committed:
-Using travis for testing (for now).

amitaibu commented:
You can add the pr to the make file
On Sep 16, 2015 5:29 PM, "Amir Arbel" notifications@github.com wrote:

> This PR requires using RESTful-Drupal/restful#642
> https://github.com/RESTful-Drupal/restful/pull/642
> Otherwise notification variable won't be sent to the client. It means
> that the notification can only be activated and the bell icon status would
> be wrong: (The toggle request depends on that variable).
> [image: selection_082]
> https://cloud.githubusercontent.com/assets/2690042/9907469/fd68a41a-5c97-11e5-9e4a-a4a3dc3957cc.jpg
> [image: selection_084]
> https://cloud.githubusercontent.com/assets/2690042/9907524/2e79d29a-5c98-11e5-8dcc-b6ced6fbc705.jpg
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/shoov/shoov/pull/219#issuecomment-140758562.

Moo64c commented:
@amitaibu 

> You can add the pr to the make file

You have an example for this? Remember I'm committing to a forked repository.

Moo64c committed:
Cleanup.

amitaibu commented:
Then get the forked repo. Other gizras may have examples
On Sep 16, 2015 6:17 PM, "Amir Arbel" notifications@github.com wrote:

> @amitaibu https://github.com/amitaibu
> 
> You can add the pr to the make file
> 
> You have an example for this? Remember I'm committing to a forked
> repository.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/shoov/shoov/pull/219#issuecomment-140773491.

Moo64c committed:
Switch repo.

Moo64c commented:
@amitaibu 
Switched the repos (for now). Build is failing because wrong data types - I couldn't test locally for some reason (crashed my laptop several times). I'll try to fix it today...
